### ISSUE TITLE
hasMore prop is being ignored

### ISSUE THREAD

**Issue by b4l4zs** [ankeetmaini/react-infinite-scroll-component#215]
The next function doesn't get called if the data length didn't change after the previous call, even though I pass true to the hasMore prop.
I want the next function to be called even if the data length didn't change after the previous call.

**Comment by KeitelDOG** [comment]
Can't get `next()` to fire. I made infinite scroll it work with 2 virtualization lists, 1 normal list. But this one seems to never call the next function. Don't know why yet. Whereas it display endMessage at the bottom when setting `hasMore={false}`.

**Comment by KeitelDOG** [comment]
I got it to work, use parent scrollable div. Maybe the ancestors containers has been in conflict with the scrolling calculations, because one of them had overflow auto already. But using a direct parent did it.

**Comment by gabe2code-opstalent** [comment]
I had the same problem and I solved passing a different key (basically remount the component) when I need the component fires the next event again. I hope that helps

**Comment by exapsy** [comment]
@gabe2code-opstalent Yeah it seems like it has something to do with rerendering. If you remove the key, you can reproduce that solution (with hot-reload) just by making a simple change (like adding a console log) so it rerenders. When it rerenders it loads the next page.

**Comment by YuryYaroshevich** [comment]
> I had the same problem and I solved passing a different key (basically remount the component) when I need the component fires the `next()` event again. I hope that helps

@gabe2code-opstalent Could you please explain what do you mean by passing different key? Do you mean `key="xx"` attribute on the element of the list?
